profile/src/main/java/com/scottlogic/deg/profile/reader/ConstraintReader.java

```diff
@@ -0,0 +1,220 @@
+    }
+
+    private Constraint readRelationalConstraintDto(RelationalConstraintDTO dto, ProfileFields fields)
+    {
+        Field main = fields.getByName(dto.field);
+        Field other = fields.getByName(dto.getOtherField());
+        switch (dto.getType())
+        {
+            case EQUAL_TO_FIELD:
+                Granularity offsetGranularity = readGranularity(main.getType(), dto.offsetUnit);
+                return offsetGranularity != null
+                        ? new EqualToOffsetRelation(main, other, offsetGranularity, dto.offset)
+                        : new EqualToRelation(main, other);
+            case AFTER_FIELD:
+                return new AfterRelation(main, other, false, DateTimeDefaults.get());
+            case AFTER_OR_AT_FIELD:
+                return new AfterRelation(main, other, true, DateTimeDefaults.get());
+            case BEFORE_FIELD:
+                return new BeforeRelation(main, other, false, DateTimeDefaults.get());
+            case BEFORE_OR_AT_FIELD:
+                return new BeforeRelation(main, other, true, DateTimeDefaults.get());
+            case GREATER_THAN_FIELD:
+                return new AfterRelation(main, other, false, NumericDefaults.get());
+            case GREATER_THAN_OR_EQUAL_TO_FIELD:
+                return new AfterRelation(main, other, true, NumericDefaults.get());
+            case LESS_THAN_FIELD:
+                return new BeforeRelation(main, other, false, NumericDefaults.get());
+            case LESS_THAN_OR_EQUAL_TO_FIELD:
+                return new BeforeRelation(main, other, true, NumericDefaults.get());
+            default:
+                throw new InvalidProfileException("Unexpected relation data type " + dto.getType());
+        }
+    }
+
+    private Constraint readAtomicConstraintDto(AtomicConstraintDTO dto, ProfileFields profileFields)
+    {
+        Field field = profileFields.getByName(dto.field);
+        switch (dto.getType())
+        {
+            case EQUAL_TO:
+                return new EqualToConstraint(field, readAnyType(field, ((EqualToConstraintDTO)dto).value));
+            case IN_SET:
+                InSetConstraintDTO inSetConstraintDTO = (InSetConstraintDTO) dto;
+                DistributedList<Object> values = (inSetConstraintDTO instanceof InSetFromFileConstraintDTO
+                        ? fileReader.setFromFile(((InSetFromFileConstraintDTO) inSetConstraintDTO).file)
+                        : DistributedList.uniform(((InSetOfValuesConstraintDTO) inSetConstraintDTO).values.stream()
+                        .distinct()
+                        .map(o -> readAnyType(field, o))
+                        .collect(Collectors.toList())));
+                return new IsInSetConstraint(field, values);
+            case IN_MAP:
+                InMapConstraintDTO inMapConstraintDTO = (InMapConstraintDTO) dto;
+                return new InMapRelation(field, profileFields.getByName(inMapConstraintDTO.file),
```

Comment:
Same as above, a little too much going on for a switch statement

Reply:
Agreed